js/api.ts

```diff
@@ -186,6 +186,17 @@ export function fetchFirestoreJWT(classHash: string, resourceLinkId: string | nu
   }
 }
 
+export function fetchFirestoreJWTWithDefaultParams(firebaseApp?: string): Promise<{token: string}> {
+  return Promise.all([fetchOfferingData(), fetchClassData()])
+    .then(([offeringData, classData]) => {
+      const resourceLinkId = offeringData.id.toString();
+      // only pass resourceLinkId if there is a studentId
+      // This could be a teacher or researcher viewing the report of a student
+      // The studentId is sent in the firestore JWT request as the target_user_id
+      return fetchFirestoreJWT(classData.class_hash, urlParam("studentId") ? resourceLinkId : null, urlParam("studentId"), firebaseApp);
```

Comment:
```suggestion
      const studentId = urlParam("studentId");
      return fetchFirestoreJWT(classData.class_hash, studentId ? resourceLinkId : null, studentId, firebaseApp);
```
I realize this is copy/pasted code, but I'd take this opportunity to DRY it up.

Reply:
Thanks, I've fixed it here: https://github.com/concord-consortium/portal-report/pull/353/commits/5dd86538980144e889a9743c4a9c2392fd51fe0d